### PR TITLE
Migrate std lib trim to guava trim

### LINKED ISSUE #5105: Migrate calls to String::trim() to Guava's whitespace trimming
<s>Java 11 introduced the `String::strip()` method, which strips whitespace around a string, including newer Unicode whitespace characters.
We currently use the `String::trim()` method in a number of places. In many of those places, the string being trimmed can legitimately contain any Unicode character, so `String::strip()` should be used instead.

Since OpenRefine now requires Java 11+, we can migrate to `String::strip()` in most places. Note that `String::trim()` apparently removes the `\u0000` character and `String::strip()` does not, but I am not sure if that is relevant anywhere for us.</s>

We should rather migrate to Guava's whitespace trimming, see [@ostephens's post below](https://github.com/OpenRefine/OpenRefine/issues/5105#issuecomment-1275219596).

### PR DESCRIPTION
**Pull request by vijal-patel**
Fixes #5105 

Changes proposed in this pull request:
- All uses of the std lib String trim have been migrated to Guava's implementation of trim

I have also tested the csv file from #5246 to verify that the NBSP's are catered for.

### CODE REVIEW

**Review by wetneb — COMMENTED** [review]
Thank you!

File: main/src/com/google/refine/importers/ImporterUtilities.java [hunk 1]
```diff
@@ -57,7 +58,7 @@ public class ImporterUtilities {
 
     static public Serializable parseCellValue(String text) {
         if (text.length() > 0) {
-            String text2 = text.trim();
+            String text2 = CharMatcher.whitespace().trimFrom(text);
```
[the remark this answers is not in the document]
Comment (vijal-patel) [reply main/src/com/google/refine/importers/ImporterUtilities.java:61]: good question, I will investigate this and get back to you

[the remark this answers is not in the document]
Comment (tfmorris) [reply main/src/com/google/refine/importers/ImporterUtilities.java:61]: It's a static instance 

https://github.com/google/guava/blob/master/guava/src/com/google/common/base/CharMatcher.java#L1211-L1225

**Review by tfmorris — APPROVED** [review]

File: main/src/com/google/refine/importers/ImporterUtilities.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (wetneb) [reply main/src/com/google/refine/importers/ImporterUtilities.java:61]: Thanks for looking into it!

**Pull request merged**